packages/react-form-state/README.md

```diff
@@ -0,0 +1,947 @@
+  submitting: boolean;
+  errors: ClientError[];
+  reset(): void;
+  submit(): void;
+}
+```
+
+The `formDetails` object passed into `children` contains a `fields` dictionary, which includes all the state and handlers you need to render your form.
+
+```typescript
+export interface FieldState<Value> {
+  name: string;
+  initialValue: Value;
+  value: Value;
+  dirty: boolean;
+  changed: boolean;
+  error?: any;
+  onBlur(): void;
+}
+```
+
+#### Example
+
+The simplest way to use this component is to pass it some `initialValues` and render some `input`s in it's `children` render prop.
+
+```typescript
+import FormState from '@shopify/react-form-state';
+
+function MyComponent() {
+  return (
+    <FormState
+      initialValues={{
+        title: 'Cool title',
+        description: 'Cool product',
+      }}
+    >
+      {formDetails => {
+        const {fields} = formDetails;
+        const {title, description} = fields;
+
+        return (
+          <form>
+            <label htmlFor={title.name}>Title</label>
+            <input
+              {...title}
+              onChange={({currentTarget}) => {
+                // our onChange expects just a value, no event needed
+                title.onChange(currentTarget.value);
+              }}
+            />
+
+            <label htmlFor={title.name}>Title</label>
+            <input
+              {...description}
+              onChange={({currentTarget}) => {
+                description.onChange(currentTarget.value);
+              }}
+            />
+          </form>
+        );
+      }}
+    </FormState>
+  );
+}
+```
+
+#### Reducing boilerplate with custom inputs
+
+The previous example has a fair amount of repetition around passing the right handlers to the right inputs. It also doesn't handle rendering errors. We can eliminate this boilerplate by using a custom component that's props match our field descriptor objects.
+
+It just so happens that [@shopify/polaris](https://polaris.shopify.com/components/forms/text-field)'s `<TextField />` component does this. So, using `<TextField />` and ES6 [spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) syntax we can greatly simplify our forms. The Polaris inputs also support `error` props, which we'll take advantage of later.
+
+#### Example
+
+```typescript
+import {TextField} from '@shopify/polaris';
+import FormState from '@shopify/react-form-state';
+
+function MyComponent() {
+  return (
+    <FormState
+      initialValues={{
+        title: 'Cool title',
+        description: 'Cool product',
+      }}
+    >
+      {formDetails => {
+        const {fields} = formDetails;
+
+        return (
+          <form>
+            <TextField label="Title" {...fields.title} />
+            <TextField label="Description" {...fields.description} />
+          </form>
+        );
+      }}
+    </FormState>
+  );
+}
+```
+
+There is also nothing to stop you from using your own custom inputs.
```

Comment:
I'd :fire: this line. They can infer that from the above sentences, and we probably do want to push polaris here, for consistency.